core/src/main/java/org/apache/carbondata/core/cache/CacheProvider.java

```diff
@@ -84,14 +84,13 @@ public static CacheProvider getInstance() {
     //check if cache is null for given cache type, if null create one
     if (!dictionaryCacheAlreadyExists(cacheType)) {
       synchronized (lock) {
-        if (!dictionaryCacheAlreadyExists(cacheType)) {
-          if (null == cacheTypeToLRUCacheMap.get(cacheType)) {
-            createLRULevelCacheInstance(cacheType);
-          }
-          createDictionaryCacheForGivenType(cacheType, carbonStorePath);
+        if (null == cacheTypeToLRUCacheMap.get(cacheType)) {
+          createLRULevelCacheInstance(cacheType);
         }
+        createDictionaryCacheForGivenType(cacheType, carbonStorePath);
```

Comment:
Yes I agree...but it can happen that multiple threads are waiting on the synchronized block and when one completes the next one enters, then for that thread also dictionary cache object is non existent and it will also create a new object.
May be you can once go through double locking mechanism and try to relate it with that.